layers/eight_mile/tf/optz.py

```diff
@@ -776,17 +776,17 @@ def weight_decay_fn():
     def global_step(self):
         return self.optimizer.iterations
 
-    def update(self, model, x, y, num_replicas=1):
+    def update(self, model, x, y, num_replicas=1, **kwargs):
```

Comment:
there doesnt seem to be much API parity between TF and PyTorch with this feature.  In once case we are changing the impl of EagerOptimizer and in the other its unmodified?

Reply:
There already wasn't API parity between the two. The pytorch seq2seq model is structured like all out original models (tf1 and pytorch) where there is a `.create_loss` method. TF2 Eager (the only seq2seq supported mode) is already setup differently where there is a `loss` function in the trainer that is responsible for running the model's forward pass and calculating the loss. That function is called by the eager optimizer so the only way to inject things that tweak that loss calculation is to pass them through the EagerOptimzer like this. The TF2 Eager version doesn't have something akin to `.create_loss` that we can modify or override in a subclass.